File: ASP/Top2000/Models/Position.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Models
{
    public class Position
    {
        [Key]
        public int Year { get; set; }

        public int Place { get; set; }

        [ForeignKey("SongId")]
        public int SongId { get; set; }
        public Song Song { get; set; } = new();
    }
}

File: Flutter/top2000/lib/generators/position.dart
import 'package:faker/faker.dart';
import 'package:jiffy/jiffy.dart';
import 'package:top2000/generators/songs.dart';
import 'package:top2000/models/Position.dart';
import 'package:top2000/models/song.dart';

Future<List<Position>> getPosition({int length = 1}) async {
  List<Position> list = List<Position>.empty(growable: true);
  final Faker faker = Faker();

  for (int i = 0; i < length; i++) {
    list.add(
      Position(
        place: faker.randomGenerator.integer(2000),
        song: await getSongs(length: 1).then((List<Song> value) => value[0]),
        year: Jiffy(faker.date.year(minYear: 1940, maxYear: Jiffy().year)).year,
      ),
    );
  }

  return list;
}
